### PR TITLE
autoconf: Add autogen.sh, remove configure and config.h.in

### PR DESCRIPTION
**Pull request by fxlb**
The minimum required version of autoconf is currently 2.69.

Put autoconf-generated files in the release tarball.

If version 2.69 or later is already installed and there is no autoconf default, it may be necessary to set the AUTORECONF environment variable to enable the one to use, like:
AUTORECONF=autoreconf-2.69 ./autogen.sh
or
AUTORECONF=autoreconf-2.71 ./autogen.sh

### CODE REVIEW

File: Makefile.in [hunk 1]
```diff
@@ -824,17 +823,48 @@ extags: $(TAGFILES)
...
 	    echo "Archive build from tag $$TAG."; \
 	else \
-	    git archive --prefix="$$TAG"/ -o "$$TAG".tar.gz HEAD \
-	    $(RELEASE_FILES) && \
+	    (git -C "$$DIR" archive --format=tar --prefix="$$TAG"/ "HEAD^{tree}" \
+	    $(RELEASE_FILES) | tar xf -) && \
 	    echo "No $$TAG tag. Archive build from HEAD."; \
-	fi
+	fi && \
+	(cd "$$TAG" && $${AUTORECONF-autoreconf} && rm -rf autom4te.cache) && \
```
Comment (infrastation) [inline Makefile.in:864]: `${VARNAME:-defaultvalue}`

File: Makefile.in [hunk 2]
```diff
@@ -824,17 +823,48 @@ extags: $(TAGFILES)
...
+	(tar cf - "$$TAG" | gzip > "$$DIR/$$TAG".tar.gz) && \
+	cd /tmp && \
+	rm -rf "$$AUTORECONF_DIR"
```
Comment (infrastation) [inline Makefile.in:867]: just `rm -rf /tmp/"$$AUTORECONF_DIR"`

**Comment by fxlb** [comment]
Updated.

**Comment by fxlb** [comment]
Rebased. Sync with tcpslice updates.

**Comment by infrastation** [comment]
Please port the most recent bug fix from tcpslice into this draft so subsequent rebuilds of this pull request do not leave temporary directories behind.

**Pull request merged**